### PR TITLE
Expose -ftime-{trace,report} build options

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

n/t

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

**Review by tobim — APPROVED** [review]

**Comment by dominiklohmann** [comment]
I ran Clang Build Analyzer on this branch for the build target `vast`. It shows a few things we still can work on.

```
Analyzing build trace from 'trace.json'...
**** Time summary:
Compilation (335 times):
  Parsing (frontend):         2165.7 s
  Codegen & opts (backend):    833.8 s

**** Files that took longest to parse (compiler frontend):
 68949 ms: build/libvast/CMakeFiles/libvast.dir/src/system/node.cpp.o
 58957 ms: build/libvast/CMakeFiles/libvast.dir/src/system/application.cpp.o
 42068 ms: build/libvast/CMakeFiles/libvast.dir/src/type.cpp.o
 41374 ms: build/libvast/CMakeFiles/libvast.dir/src/system/application.cpp.o
 37942 ms: build/libvast/CMakeFiles/libvast.dir/src/system/partition.cpp.o
 35795 ms: build/libvast/CMakeFiles/libvast.dir/src/system/node.cpp.o
 35623 ms: build/libvast/CMakeFiles/libvast.dir/src/value_index_factory.cpp.o
 35584 ms: build/libvast/CMakeFiles/libvast.dir/src/system/index.cpp.o
 28445 ms: build/libvast/CMakeFiles/libvast.dir/src/system/exporter.cpp.o
 25075 ms: build/libvast/CMakeFiles/libvast.dir/src/concept/parseable/vast/expression.cpp.o

**** Files that took longest to codegen (compiler backend):
 67469 ms: build/libvast/CMakeFiles/libvast.dir/src/system/node.cpp.o
 55545 ms: build/libvast/CMakeFiles/libvast.dir/src/system/application.cpp.o
 36497 ms: build/libvast/CMakeFiles/libvast.dir/src/concept/parseable/vast/expression.cpp.o
 30759 ms: build/libvast/CMakeFiles/libvast.dir/src/system/partition.cpp.o
 29817 ms: build/libvast/CMakeFiles/libvast.dir/src/system/index.cpp.o
 25209 ms: build/libvast/CMakeFiles/libvast.dir/src/system/exporter.cpp.o
 23875 ms: build/libvast/CMakeFiles/libvast.dir/src/system/type_registry.cpp.o
 23591 ms: build/libvast/CMakeFiles/libvast.dir/src/value_index_factory.cpp.o
 18421 ms: build/libvast/CMakeFiles/libvast.dir/src/type.cpp.o
 18283 ms: build/libvast/CMakeFiles/libvast.dir/src/view.cpp.o

**** Templates that took longest to instantiate:
 69486 ms: caf::anon_send_exit<caf::intrusive_ptr<caf::actor_control_block>> (98 times, avg 709 ms)
 69276 ms: caf::make_message<caf::exit_msg> (98 times, avg 706 ms)
 68270 ms: caf::make_counted<caf::detail::tuple_vals<caf::exit_msg>, caf::exit_... (98 times, avg 696 ms)
 63369 ms: caf::inspect<caf::deserializer> (546 times, avg 116 ms)
 54452 ms: vast::detail::hash_inspector<vast::xxhash64>::operator()<caf::meta::... (200 times, avg 272 ms)
 51871 ms: caf::inspect<caf::serializer> (533 times, avg 97 ms)
 40890 ms: std::__1::__function::__func<(lambda at ../libvast/aux/caf/libcaf_co... (612 times, avg 66 ms)
 35379 ms: std::__1::function<caf::error (caf::meta::type_name_t, unsigned char... (204 times, avg 173 ms)
 35190 ms: std::__1::__function::__value_func<caf::error (caf::meta::type_name_... (204 times, avg 172 ms)
 34976 ms: std::__1::__function::__value_func<caf::error (caf::meta::type_name_... (204 times, avg 171 ms)
 31863 ms: caf::inspect<caf::deserializer, caf::none_t, vast::conjunction, vast... (20 times, avg 1593 ms)
 29729 ms: caf::inspect<vast::detail::hash_inspector<vast::xxhash64>, caf::none... (194 times, avg 153 ms)
 28910 ms: caf::detail::tuple_vals_impl<caf::detail::message_data, caf::exit_ms... (98 times, avg 295 ms)
 28858 ms: caf::detail::tuple_vals_impl<caf::detail::message_data, caf::exit_ms... (98 times, avg 294 ms)
 27691 ms: vast::uhash<vast::xxhash64>::operator()<vast::type> (100 times, avg 276 ms)
 27392 ms: caf::detail::tuple_inspect_delegate<0, 0, std::__1::tuple<caf::exit_... (98 times, avg 279 ms)
 27314 ms: caf::data_processor<caf::deserializer>::operator()<caf::exit_msg &> (98 times, avg 278 ms)
 27051 ms: vast::detail::hash_inspector<vast::xxhash64>::operator()<signed char... (100 times, avg 270 ms)
 26881 ms: caf::inspect<caf::deserializer, vast::attribute_extractor, vast::fie... (20 times, avg 1344 ms)
 26829 ms: caf::data_processor<caf::deserializer>::apply<caf::exit_msg> (98 times, avg 273 ms)
 26775 ms: caf::data_processor<caf::deserializer>::operator()<caf::meta::type_n... (98 times, avg 273 ms)
 26591 ms: vast::detail::hash_inspector<vast::xxhash64>::operator()<vast::type:... (100 times, avg 265 ms)
 26445 ms: caf::detail::tuple_vals_impl<caf::detail::message_data, caf::exit_ms... (98 times, avg 269 ms)
 26403 ms: caf::detail::tuple_vals_impl<caf::detail::message_data, caf::exit_ms... (98 times, avg 269 ms)
 25772 ms: caf::visit<vast::detail::hash_inspector<vast::xxhash64> &, vast::typ... (100 times, avg 257 ms)
 25706 ms: caf::visit_impl<void, 1>::apply<vast::detail::hash_inspector<vast::x... (100 times, avg 257 ms)
 25668 ms: caf::sum_type_access<vast::type>::apply<void, caf::visit_impl_contin... (100 times, avg 256 ms)
 25609 ms: caf::sum_type_access<vast::type>::dispatcher<void, caf::visit_impl_c... (100 times, avg 256 ms)
 24874 ms: caf::detail::tuple_inspect_delegate<0, 0, std::__1::tuple<caf::exit_... (98 times, avg 253 ms)
 24793 ms: caf::type_erased_value_factory::operator()<caf::exit_msg &> (98 times, avg 252 ms)

**** Template sets that took longest to instantiate:
168196 ms: caf::make_counted<$> (1373 times, avg 122 ms)
167704 ms: caf::detail::tuple_vals_impl<$>::dispatch<$> (3370 times, avg 49 ms)
153604 ms: caf::make_message<$> (821 times, avg 187 ms)
148200 ms: caf::variant<$>::apply<$> (1002 times, avg 147 ms)
147777 ms: caf::variant<$>::apply_impl<$> (1002 times, avg 147 ms)
142085 ms: caf::detail::tuple_inspect_delegate<$> (24683 times, avg 5 ms)
130087 ms: std::__1::function<$>::function<$> (751 times, avg 173 ms)
129228 ms: std::__1::__function::__value_func<$>::__value_func<$> (751 times, avg 172 ms)
127774 ms: caf::visit<$> (283 times, avg 451 ms)
127579 ms: caf::visit_impl<$>::apply<$> (283 times, avg 450 ms)
126737 ms: caf::inspect<$> (787 times, avg 161 ms)
107906 ms: std::__1::__function::__func<$>::__func (751 times, avg 143 ms)
102344 ms: caf::data_processor<$>::apply<$> (814 times, avg 125 ms)
 91584 ms: std::__1::__function::__alloc_func<$>::__alloc_func (2253 times, avg 40 ms)
 91301 ms: std::__1::forward_as_tuple<$> (3686 times, avg 24 ms)
 87427 ms: std::__1::decay<$> (77458 times, avg 1 ms)
 80574 ms: caf::sum_type_access<$>::apply<$> (207 times, avg 389 ms)
 78753 ms: caf::sum_type_access<$>::dispatcher<$>::dispatch<$> (205 times, avg 384 ms)
 76653 ms: caf::sum_type_access<$>::dispatcher<$>::invoke<$> (3280 times, avg 23 ms)
 73241 ms: caf::detail::apply_args_suffxied<$> (3263 times, avg 22 ms)
 70733 ms: caf::make_mailbox_element<$> (173 times, avg 408 ms)
 69505 ms: caf::anon_send_exit<$> (108 times, avg 643 ms)
 68619 ms: caf::detail::tuple_vals_impl<$>::load (674 times, avg 101 ms)
 67596 ms: std::__1::tuple<$> (6441 times, avg 10 ms)
 67373 ms: caf::mailbox_element_vals<$>::mailbox_element_vals<$> (173 times, avg 389 ms)
 67096 ms: caf::detail::profiled_send<$> (200 times, avg 335 ms)
 65663 ms: vast::system::make_source<$> (16 times, avg 4103 ms)
 64550 ms: caf::default_sum_type_access<$>::apply<$> (170 times, avg 379 ms)
 60779 ms: std::__1::unordered_map<$> (1263 times, avg 48 ms)
 56327 ms: std::__1::__hash_table<$> (1380 times, avg 40 ms)

**** Functions that took longest to compile:
   297 ms: bool vast::choice_parser<vast::action_parser<vast::sequence_parser<v... (../libvast/src/concept/parseable/vast/expression.cpp)
   220 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
   106 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/application.cpp)
   100 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/application.cpp)
    92 ms: vast::detail::eval_result_type<vast::bitmap, vast::bitmap>::type vas... (../libvast/src/value_index.cpp)
    88 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/application.cpp)
    86 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
    86 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
    85 ms: vast::detail::eval_result_type<vast::bitmap, vast::bitmap>::type vas... (../libvast/src/value_index.cpp)
    85 ms: std::__1::__vector_base<caf::config_value, std::__1::allocator<caf::... (../libvast/src/system/explorer.cpp)
    80 ms: vast::detail::eval_result_type<vast::bitmap, vast::bitmap>::type vas... (../libvast/src/system/evaluator.cpp)
    76 ms: bool vast::type_parser::parse<std::__1::__wrap_iter<char*>, vast::ty... (../libvast/src/system/spawn_arguments.cpp)
    74 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/application.cpp)
    74 ms: decltype((((*(fp)) , ((fp)++)) , ((fp) == (fp0))) , ((bool)())) vast... (../libvast/src/format/zeek.cpp)
    73 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
    71 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
    70 ms: vast::format::json::reader<vast::format::json::field_selector<vast::... (../libvast/src/system/application.cpp)
    70 ms: vast::detail::eval_result_type<vast::bitmap, vast::bitmap>::type vas... (../libvast/src/system/evaluator.cpp)
    69 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
    69 ms: vast::system::node_state::spawn_command(vast::invocation const&, caf... (../libvast/src/system/node.cpp)
    68 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/node.cpp)
    68 ms: vast::system::index(caf::stateful_actor<vast::system::index_state, c... (../libvast/src/system/index.cpp)
    67 ms: vast::detail::eval_result_type<vast::bitmap, vast::bitmap>::type vas... (../libvast/src/system/eraser.cpp)
    67 ms: vast::detail::eval_result_type<vast::bitmap, vast::bitmap>::type vas... (../libvast/src/system/index.cpp)
    67 ms: caf::expected<vast::system::(anonymous namespace)::make_source_resul... (../libvast/src/system/application.cpp)
    67 ms: bool vast::type_parser::parse<char const*, vast::type>(char const*&,... (../libvast/src/format/test.cpp)
    67 ms: decltype(fp0.out()) fmt::v7::formatter<fmt::v7::basic_string_view<ch... (../libvast/src/system/partition.cpp)
    66 ms: vast::format::zeek::reader::parse_header() (../libvast/src/format/zeek.cpp)
    66 ms: vast::system::(anonymous namespace)::make_root_command(std::__1::bas... (../libvast/src/system/application.cpp)
    66 ms: vast::system::read_query(vast::invocation const&, std::__1::basic_st... (../libvast/src/system/read_query.cpp)

**** Function sets that took longest to compile / optimize:
  9697 ms: decltype(fp0(decltype(std::__1::__declval<$>(0)) std::__1::declval<$... (2965 times, avg 3 ms)
  4262 ms: auto caf::make_dispatch_fun<$>() (2801 times, avg 1 ms)
  3365 ms: vast::choice_parser<$>::choice_parser(vast::choice_parser<$>, vast::... (708 times, avg 4 ms)
  3000 ms: std::__1::__compressed_pair<$>::__compressed_pair<$>(std::__1::piece... (520 times, avg 5 ms)
  2941 ms: caf::trivial_match_case<$>::invoke(caf::detail::invoke_result_visito... (525 times, avg 5 ms)
  2636 ms: std::__1::enable_if<$>::type vast::to_parser<$>(std::__1::__wrap_ite... (1549 times, avg 1 ms)
  2486 ms: caf::deserializer::result_type caf::inspect<$>(caf::deserializer&, c... (78 times, avg 31 ms)
  2207 ms: caf::intrusive_ptr<$>::~intrusive_ptr() (1473 times, avg 1 ms)
  2098 ms: caf::detail::tuple_vals_impl<$>::stringify(unsigned long) const (594 times, avg 3 ms)
  2023 ms: decltype(fp0(std::get<$>(fp))) caf::detail::tuple_inspect_delegate<$... (2540 times, avg 0 ms)
  2023 ms: decltype(fp0(std::get<$>(fp))) caf::detail::tuple_inspect_delegate<$... (2519 times, avg 0 ms)
  1988 ms: decltype(fp0(std::get<$>(fp))) caf::detail::tuple_inspect_delegate<$... (2501 times, avg 0 ms)
  1848 ms: std::__1::__vector_base<$>::__vector_base() (484 times, avg 3 ms)
  1715 ms: caf::detail::tuple_vals_impl<$>::get(unsigned long) const (593 times, avg 2 ms)
  1695 ms: std::__1::__vector_base<$>::__vector_base(std::__1::allocator<$>&&) (487 times, avg 3 ms)
  1694 ms: auto vast::parser<$>::then<$>(std::__1::__wrap_iter<$>) const (813 times, avg 2 ms)
  1676 ms: std::__1::vector<$>::max_size() const (555 times, avg 3 ms)
  1609 ms: void caf::detail::default_behavior_impl<$>::init<$>(std::__1::integr... (661 times, avg 2 ms)
  1533 ms: std::__1::enable_if<$>::type std::__1::__split_buffer<$>::__construc... (290 times, avg 5 ms)
  1514 ms: std::__1::__compressed_pair_elem<$>::__compressed_pair_elem<$>(std::... (745 times, avg 2 ms)
  1433 ms: std::__1::vector<$>::_ConstructTransaction::~_ConstructTransaction() (556 times, avg 2 ms)
  1424 ms: std::__1::__split_buffer<$>::__split_buffer(unsigned long, unsigned ... (399 times, avg 3 ms)
  1388 ms: std::__1::vector<$>::__recommend(unsigned long) const (473 times, avg 2 ms)
  1370 ms: caf::error caf::data_processor<$>::fill_range<$>(std::__1::vector<$>... (221 times, avg 6 ms)
  1336 ms: caf::intrusive_ptr<$>::detach() (1337 times, avg 0 ms)
  1286 ms: std::__1::vector<$>::vector(std::__1::vector<$> const&) (362 times, avg 3 ms)
  1274 ms: caf::detail::tuple_vals_impl<$>::type(unsigned long) const (593 times, avg 2 ms)
  1239 ms: vast::sequence_printer<$>::sequence_printer(vast::and_printer<$>, va... (306 times, avg 4 ms)
  1236 ms: std::__1::vector<$>::__annotate_delete() const (720 times, avg 1 ms)
  1215 ms: void caf::variant<$>::apply_impl<$>(caf::variant<$>&, caf::detail::v... (233 times, avg 5 ms)

*** Expensive headers:
184396 ms: ../libvast/vast/logger.hpp (included 92 times, avg 2004 ms), included via:
  logger.cpp.o  (4809 ms)
  type.cpp.o  (3937 ms)
  remote_command.cpp.o  (3907 ms)
  taxonomies.cpp.o  (3903 ms)
  value_index_factory.cpp.o  (3822 ms)
  spawn_or_connect_to_node.cpp.o  (3674 ms)
  ...

147155 ms: ../libvast/vast/fwd.hpp (included 152 times, avg 968 ms), included via:
  view.cpp.o view.hpp aliases.hpp  (2226 ms)
  node.cpp.o node.hpp aliases.hpp  (2147 ms)
  shutdown.cpp.o shutdown.hpp defaults.hpp  (2135 ms)
  sink.cpp.o sink.hpp  (2092 ms)
  value_index.cpp.o value_index.hpp error.hpp  (2076 ms)
  type.cpp.o type.hpp  (2056 ms)
  ...

49129 ms: /usr/local/opt/llvm/include/c++/v1/algorithm (included 166 times, avg 295 ms), included via:
  wah_bitmap.cpp.o iostream ios __locale string string_view __string  (1154 ms)
  time_synopsis.cpp.o time_synopsis.hpp min_max_synopsis.hpp deserializer.hpp string string_view __string  (1111 ms)
  remote_command.cpp.o remote_command.hpp command.hpp string.hpp  (1083 ms)
  pivot_command.cpp.o pivot_command.hpp command.hpp string.hpp  (1077 ms)
  pcap_writer_command.cpp.o pcap_writer_command.hpp command.hpp string.hpp  (1011 ms)
  infer_command.cpp.o infer_command.hpp command.hpp string.hpp  (991 ms)
  ...

46039 ms: ../libvast/vast/command.hpp (included 39 times, avg 1180 ms), included via:
  remote_command.cpp.o remote_command.hpp  (3461 ms)
  pivot_command.cpp.o pivot_command.hpp  (3071 ms)
  pcap_writer_command.cpp.o pcap_writer_command.hpp  (3062 ms)
  infer_command.cpp.o infer_command.hpp  (2933 ms)
  read_query.cpp.o read_query.hpp  (2836 ms)
  version_command.cpp.o version_command.hpp  (2655 ms)
  ...

44948 ms: ../libvast/vast/type.hpp (included 100 times, avg 449 ms), included via:
  type.cpp.o  (3373 ms)
  taxonomies.cpp.o taxonomies.hpp type_set.hpp  (1407 ms)
  table_slice_column.cpp.o table_slice_column.hpp qualified_record_field.hpp  (1319 ms)
  time_synopsis.cpp.o time_synopsis.hpp min_max_synopsis.hpp synopsis.hpp  (1089 ms)
  table_slice_row.cpp.o table_slice_row.hpp table_slice.hpp  (1075 ms)
  partition_synopsis.cpp.o partition_synopsis.hpp qualified_record_field.hpp  (1010 ms)
  ...

42531 ms: ../libvast/vast/concept/printable/vast/data.hpp (included 22 times, avg 1933 ms), included via:
  taxonomies.cpp.o  (2856 ms)
  partition.cpp.o expression.hpp  (2654 ms)
  indexer.cpp.o expression.hpp  (2369 ms)
  index.cpp.o expression.hpp  (2352 ms)
  pivoter.cpp.o expression.hpp  (2266 ms)
  explorer.cpp.o expression.hpp  (2116 ms)
  ...

41586 ms: ../libvast/aux/caf/libcaf_core/caf/actor_system.hpp (included 101 times, avg 411 ms), included via:
  signal_monitor.cpp.o signal_monitor.hpp  (1418 ms)
  defaults.cpp.o  (1106 ms)
  plugin.cpp.o plugin.hpp actor_system_config.hpp actor_factory.hpp  (932 ms)
  start_command.cpp.o actor_system_config.hpp actor_factory.hpp  (922 ms)
  default_configuration.cpp.o default_configuration.hpp configuration.hpp actor_system_config.hpp actor_factory.hpp  (844 ms)
  configuration.cpp.o configuration.hpp actor_system_config.hpp actor_factory.hpp  (808 ms)
  ...

39107 ms: ../libvast/vast/error.hpp (included 119 times, avg 328 ms), included via:
  value_index.cpp.o value_index.hpp  (3285 ms)
  logger.cpp.o logger.hpp  (2025 ms)
  error.cpp.o  (1835 ms)
  expression_visitors.cpp.o expression_visitors.hpp  (1746 ms)
  ostream_writer.cpp.o ostream_writer.hpp  (1564 ms)
  operator.cpp.o operator.hpp  (1266 ms)
  ...

37640 ms: ../libvast/aux/caf/libcaf_core/caf/scheduled_actor.hpp (included 98 times, avg 384 ms), included via:
  exporter.cpp.o exporter.hpp  (1457 ms)
  fill_status_map.cpp.o  (623 ms)
  taxonomies.cpp.o logger.hpp logger_formatters.hpp event_based_actor.hpp  (596 ms)
  shutdown.cpp.o shutdown.hpp event_based_actor.hpp  (565 ms)
  type.cpp.o logger.hpp logger_formatters.hpp event_based_actor.hpp  (560 ms)
  value_index_factory.cpp.o logger.hpp logger_formatters.hpp event_based_actor.hpp  (556 ms)
  ...

36960 ms: ../libvast/vast/data.hpp (included 97 times, avg 381 ms), included via:
  test.cpp.o test.hpp  (2067 ms)
  view.cpp.o view.hpp  (1937 ms)
  data.cpp.o  (1797 ms)
  time.cpp.o  (1624 ms)
  table_slice_builder.cpp.o table_slice_builder.hpp view.hpp  (1208 ms)
  importer.cpp.o importer.hpp  (1190 ms)
  ...
```

**Comment by mavam** [comment]
cc @JoeLoser

File: configure [hunk 1]
```diff
@@ -159,6 +159,9 @@ while [ $# -ne 0 ]; do
     --show-time-report)
       append_cache_entry VAST_ENABLE_TIME_REPORT BOOL yes
       ;;
+    --show-time-trace)
+      append_cache_entry VAST_ENABLE_TIME_TRACE BOOL yes
+      ;;
```
Comment (mavam) [inline configure:164]: Please also update the usage help.

Comment (dominiklohmann) [reply configure:164]: Such a nuisance that we need to mirror this manually from CMake. I always forget this. Will do.

Comment (mavam) [reply configure:164]: Feel free to write shell function that generates the help automatically. 😉

Comment (dominiklohmann) [reply configure:165]: I'd rather get rid of the configure script entirely. It's an antipattern imo.

**Pull request merged**